lib/media/selected-text-reader.ts

```diff
@@ -0,0 +1,259 @@
+import { spawn } from 'child_process'
+import { platform, arch } from 'os'
+import { getNativeBinaryPath } from './native-interface'
+import log from 'electron-log'
+import { EventEmitter } from 'events'
+interface SelectedTextOptions {
+  format?: 'json' | 'text' // Output format
+  maxLength?: number // Maximum length of text to return
+}
+
+interface SelectedTextResult {
+  success: boolean
+  text: string | null
+  error: string | null
+  length: number
+}
+
+interface SelectedTextCommand {
+  command: 'get-text'
+  format?: 'json' | 'text'
+  maxLength?: number
+  requestId: string
+}
+
+const nativeModuleName = 'selected-text-reader'
+
+class SelectedTextReaderService extends EventEmitter {
+  #selectedTextProcess: ReturnType<typeof spawn> | null = null
```

Comment:
Do we need this `#` signs? Can we not use `private`? Or is there a deeper reason why these members need to be enforced private at runtime?

Reply:
I added it because I figured it couldn't hurt 🤷‍♂️